app/scripts/router.js

```diff
@@ -156,6 +157,20 @@ function (
       this.navigate(url, { trigger: true, replace: true });
     },
 
+    /**
+     * Redirect the user to the best suitable OAuth flow
+     */
+    redirectToBestOAuthChoice: function () {
+      var account = this.user.getChooserAccount();
+      var route = '/oauth/signup';
+
+      if (! account.isEmpty()) {
```

Comment:
Should be fine, but maybe flip sign in/up and get rid of the "not" so it's easier to read.